### DISCUSSION TITLE
How to define manytomany in multi tables?

### DISCUSSION THREAD

**Discussion by hiscaler** [cycle/orm#399] [Q&A]
I hava three tables, like

User
id, username

Product
id, title

Article
id, title

In my cast, i hava a favorite action. 
I need to record the relationship between product collections and article collections in a single table, How to i do?

**Comment by roxblnfk** [comment]
Is the User entity necessary there?

I got information about the `Product` and the `Article` entities should be stored in the same table.
Bot i dont understand how they should be related.

```
User has many Product
User has many Article
```

or 

```
Product has many Article
```
?

**Reply by hiscaler** [reply]
Is
```
User has many Product
User has many Article
```
use can favorite product or article
Current i use 
```
trait DocFavoriteTrait
{

    #[MorphedHasOne(target: DocFavorite::class, cascade: true, nullable: false, outerKey: 'doc_id', morphKey: 'doc_type')]
    protected ?DocFavorite $docFavorite = null;

    public function getDocFavorite(): ?DocFavorite
    {
        return $this->docFavorite;
    }

    public function addDocFavorite(DocFavorite $docFavorite): void
    {
        $this->docFavorite = $docFavorite;
    }

    public function removeFavorite(): void
    {
        $this->docFavorite = null;
    }

}
```
looks is ok
